src/sortStudents.ts

```diff
@@ -1,16 +1,61 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(students: Student) : number {
+  const gradesLength: number = students.grades.length;
+  const averageGrade: number = students.grades.reduce(
+    (total, num) => total + num, 0,
+  );
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return averageGrade / gradesLength;
+}
+
+export function sortStudents(students:Student[],
+  sortBy: SortType, order:SortOrder):Student[] {
+  const copyStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return copyStudents.sort((a, b) => a.name.localeCompare(b.name));
+
+    case SortType.Surname:
+      return copyStudents.sort((a, b) => a.surname.localeCompare(b.surname));
```

Comment:
let's improve your code:
these two conditions are similar, so we can do it like this:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return copyStudents.sort((a, b) => a[sortBy].localeCompare(b[sortBy]));
```
and change the SortType enum

```suggestion
    enum SortType {
      Name = 'name',
      Surname = 'surname',
      Age = 'age',
      Married = 'married',
      AverageGrade = 'averageGrade'
    }
```